### ISSUE TITLE
Save all attachments with a keybinding

### ISSUE THREAD

**Issue by wgmayer0** [neomutt/neomutt#3200]
Is it possible to save ALL attachments on an email by pressing a key. I'd like to press 's' and save ALL attachments associated with that particular email to `$HOME/downloads` without any prompts or questions.

**Comment by gahr** [comment]
We could add a `<save-all>` command, or a `<tag-all>` command so you could add a macro that does `<tag-all><tag-prefix><save-entry>`, but you might still get questions: what if the directory doesn't exist, what if an attachment doesn't have a filename, what if a filename with the same name already exists. These are all situations where we wouldn't know what to do. Perhaps we could save to a unique subdirectory... how would you approach this?

**Comment by wgmayer0** [comment]
Could there be a place where I could set my directory for everytime I want to save a file? If that directory doesn't exist, throw an error and exit advising the user to set that directory in muttrc. If the filename already exists it can number the second one with a `_2` appended to it and so forth.

**Comment by gahr** [comment]
Ok, you can set `attach_save_dir` and `attach_save_without_prompt`. You can tag the attachments you want to save, then `<tag-prefix><save-entry>`. You'll get an error if the directory does not exist, but you still get a prompt if the file already exists.

**Comment by wgmayer0** [comment]
Thanks - is this live yet? Here is my muttrc:
```
# auto download attachments
set attach_save_dir=/home/wgm/downloads
set attach_save_without_prompt=yes
```

When I launch mutt:
`Error in /home/wgm/.mutt/muttrc, line 66: attach_save_without_prompt: unknown variable`

I just updated via pacman on Arch Linux but did not see NeoMutt on the list of updated packages.

**Comment by wgmayer0** [comment]
> Ok, you can set `attach_save_dir` and `attach_save_without_prompting`. You can tag the attachments you want to save, then `<tag-prefix><save-entry>`. You'll get an error if the directory does not exist, but you still get a prompt if the file already exists.

Also I am curious if I can download all attachments from the index screen without even opening the attachments screen and tagging attachments. I plan on using my file manager to handle manipulating the files once downloaded (deleting, moving, renaming, etc) - the Unix way!

**Comment by gahr** [comment]
Typo from my side: `attach_save_without_prompting`, with *ing*

**Comment by wgmayer0** [comment]
Ok I configured these settings but I am still prompted with:
-File is a directory, save under it? ([yes]/no):
-File under directory:

How can I bypass these prompts? I would like it to save ALL attachments without any questions or details.

**Comment by gahr** [comment]
You get those prompts for attachments without a filename. They should be displayed as `<no description>` in NeoMutt's attachments view. Would you prefer neomutt to come up with random names for them?

**Comment by wgmayer0** [comment]
Yes, if possible.

**Comment by gahr** [comment]
One last clarification: when you ask about saving all attachments, do you mean all individual parts of a MIME message or all parts with "content-disposition: attachment" ?

Then, what's missing is:

* be able to save all attachments (definition to be clarified as per above) directly from the index / pager
* automatically append a suffix to existing files to avoid clashes (e.g., a `_<num>` before the last `.`)
* automatically name attachments without a content-disposition filename parameter

Does this make sense?

**Comment by wgmayer0** [comment]
All individual parts of a mime message yes so it will save the email too.
Yes that makes perfect sense thank you

**Comment by gahr** [comment]
One more clarification: this won't make html emails with referenced images in separated MIME Parts automatically work when saved to disk. Is that what you're looking for?

**Comment by dcpurton** [comment]
@wgmayer0 what's your actual end goal in saving all these files?

**Comment by wgmayer0** [comment]
The end goal is to be able to quickly save attachments with minimal fuss

**Comment by wgmayer0** [comment]
It's just way faster than answering prompts and if I can combine it with
tagging I could download all emails and attachments for
archiving/processing/piping etc and vifm is a lot better for file
management than mutt

**Comment by dcpurton** [comment]
Your goal is to archive email? Why not use a dedicated tool like `offlineimap`? Then write a Python script that processes ask the emails at once and dumps out all the attachments if you want them.

**Comment by wgmayer0** [comment]
I have notmuch script that will query my email and return it to a new mutt
window and I would like to dump those attachments to the download folder

**Comment by gahr** [comment]
Yeah I think it's very reasonable to be able to download all MIME parts at once.

**Comment by wgmayer0** [comment]
Any ideas?

**Comment by gahr** [comment]
I think this makes sense to do. @dcpurton is working on related code / ideas, and I don't want to start working on the same code :)
